src/components/shared/navigation/sidebar/VerticalMenu.tsx

```diff
@@ -217,8 +242,87 @@ export const VerticalMenuComponent: React.FC<any & { history: History }> = ({
   const [isCollapsed, setCollapsed] = useState(
     () => localStorage.getItem('hedvig:menu:collapse') === 'true',
   )
-  const [updateCount, setUpdateCount] = useState(0)
+  const [currentLocation, setCurrentLocation] = useState<string>(
+    history.location.pathname,
+  )
+  const [locations, setLocations] = useState<string[]>([])
+  const [latestClaim, setLatestClaim] = useState<LatestClaim | null>(null)
   const { isDarkmode, setIsDarkmode } = useContext(DarkmodeContext)
+  const optionIsPressed = useKeyIsPressed(KeyCode.Option)
+  const pressedKey = usePressedKey()
+
+  React.useEffect(() => {
+    history.listen((location) => {
+      setCurrentLocation(location.pathname)
+    })
+  }, [])
+
+  React.useEffect(() => {
+    const newLocations = [...locations]
+    newLocations.push(currentLocation)
+    if (newLocations.length > 10) {
+      newLocations.shift()
+    }
```

Comment:
OK feel very free to just ignore this since it's not specifically React related but would this make it so you don't have to reverse the array on line 271? You only get the last 10 locations, in the order you add them:
```suggestion
    const latestLocations = [currentLocation, ...locations]
    const newLocations = newLocations.filter((_, index) => index < 10)
```